Quantization docs: rewrite API reference to be more automated

vkuzo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #66222 Quantization docs: add pages for Numeric Suite (Eager and FX)
* #66210 Quantization documentation: move backend section down
* **#66201 Quantization docs: rewrite API reference to be more automated**
* #66198 Quantization docs: consilidate all API references on a single page
* #66129 Create a documentation page for `torch.ao.quantization.QConfig`
* #66125 Create separate documentation pages for quantization observers and fake_quants
* #66122 Create a documentation page for FX graph mode quantization APIs

Description:

This PR switches the quantization API reference to use `autosummary`
for each section.  We define the sections and manually write a list
of modules/functions/methods to include, and sphinx does the rest.
A result is a single page where we have every quantization function
and module with a quick autogenerated blurb, and user can click
through to each of them for a full documentation page.

This mimics how the `torch.nn` and `torch.nn.functional` doc
pages are set up.

In detail, for each section before this PR:
* creates a new section using `autosummary`
* adds all modules/functions/methods which were previously in the manual section
* adds any additional modules/functions/methods which are public facing but not previously documented
* deletes the old manual summary and all links to it

Test plan:

```
cd docs
make html
python -m http.server
// renders well, links work
```

Differential Revision: [D31447615](https://our.internmc.facebook.com/intern/diff/D31447615)

vkuzo committed:
Quantization docs: rewrite API reference to be more automated

Description:

This PR switches the quantization API reference to use `autosummary`
for each section.  We define the sections and manually write a list
of modules/functions/methods to include, and sphinx does the rest.
A result is a single page where we have every quantization function
and module with a quick autogenerated blurb, and user can click
through to each of them for a full documentation page.

This mimics how the `torch.nn` and `torch.nn.functional` doc
pages are set up.

In detail, for each section before this PR:
* creates a new section using `autosummary`
* adds all modules/functions/methods which were previously in the manual section
* adds any additional modules/functions/methods which are public facing but not previously documented
* deletes the old manual summary and all links to it

Test plan:

```
cd docs
make html
python -m http.server
// renders well, links work
```

[ghstack-poisoned]

jerryzh168 reviewed (approved):
great, this is much better than before!

vkuzo committed:
Update on "Quantization docs: rewrite API reference to be more automated"


Description:

This PR switches the quantization API reference to use `autosummary`
for each section.  We define the sections and manually write a list
of modules/functions/methods to include, and sphinx does the rest.
A result is a single page where we have every quantization function
and module with a quick autogenerated blurb, and user can click
through to each of them for a full documentation page.

This mimics how the `torch.nn` and `torch.nn.functional` doc
pages are set up.

In detail, for each section before this PR:
* creates a new section using `autosummary`
* adds all modules/functions/methods which were previously in the manual section
* adds any additional modules/functions/methods which are public facing but not previously documented
* deletes the old manual summary and all links to it

Test plan:

```
cd docs
make html
python -m http.server
// renders well, links work
```

Differential Revision: [D31447615](https://our.internmc.facebook.com/intern/diff/D31447615)

[ghstack-poisoned]

vkuzo committed:
Update on "Quantization docs: rewrite API reference to be more automated"


Description:

This PR switches the quantization API reference to use `autosummary`
for each section.  We define the sections and manually write a list
of modules/functions/methods to include, and sphinx does the rest.
A result is a single page where we have every quantization function
and module with a quick autogenerated blurb, and user can click
through to each of them for a full documentation page.

This mimics how the `torch.nn` and `torch.nn.functional` doc
pages are set up.

In detail, for each section before this PR:
* creates a new section using `autosummary`
* adds all modules/functions/methods which were previously in the manual section
* adds any additional modules/functions/methods which are public facing but not previously documented
* deletes the old manual summary and all links to it

Test plan:

```
cd docs
make html
python -m http.server
// renders well, links work
```

Differential Revision: [D31447615](https://our.internmc.facebook.com/intern/diff/D31447615)

[ghstack-poisoned]

vkuzo committed:
Update on "Quantization docs: rewrite API reference to be more automated"


Description:

This PR switches the quantization API reference to use `autosummary`
for each section.  We define the sections and manually write a list
of modules/functions/methods to include, and sphinx does the rest.
A result is a single page where we have every quantization function
and module with a quick autogenerated blurb, and user can click
through to each of them for a full documentation page.

This mimics how the `torch.nn` and `torch.nn.functional` doc
pages are set up.

In detail, for each section before this PR:
* creates a new section using `autosummary`
* adds all modules/functions/methods which were previously in the manual section
* adds any additional modules/functions/methods which are public facing but not previously documented
* deletes the old manual summary and all links to it

Test plan:

```
cd docs
make html
python -m http.server
// renders well, links work
```

Differential Revision: [D31447615](https://our.internmc.facebook.com/intern/diff/D31447615)

[ghstack-poisoned]

vkuzo committed:
Update on "Quantization docs: rewrite API reference to be more automated"


Description:

This PR switches the quantization API reference to use `autosummary`
for each section.  We define the sections and manually write a list
of modules/functions/methods to include, and sphinx does the rest.
A result is a single page where we have every quantization function
and module with a quick autogenerated blurb, and user can click
through to each of them for a full documentation page.

This mimics how the `torch.nn` and `torch.nn.functional` doc
pages are set up.

In detail, for each section before this PR:
* creates a new section using `autosummary`
* adds all modules/functions/methods which were previously in the manual section
* adds any additional modules/functions/methods which are public facing but not previously documented
* deletes the old manual summary and all links to it

Test plan:

```
cd docs
make html
python -m http.server
// renders well, links work
```

Differential Revision: [D31447615](https://our.internmc.facebook.com/intern/diff/D31447615)

[ghstack-poisoned]

vkuzo committed:
Update on "Quantization docs: rewrite API reference to be more automated"


Description:

This PR switches the quantization API reference to use `autosummary`
for each section.  We define the sections and manually write a list
of modules/functions/methods to include, and sphinx does the rest.
A result is a single page where we have every quantization function
and module with a quick autogenerated blurb, and user can click
through to each of them for a full documentation page.

This mimics how the `torch.nn` and `torch.nn.functional` doc
pages are set up.

In detail, for each section before this PR:
* creates a new section using `autosummary`
* adds all modules/functions/methods which were previously in the manual section
* adds any additional modules/functions/methods which are public facing but not previously documented
* deletes the old manual summary and all links to it

Test plan:

```
cd docs
make html
python -m http.server
// renders well, links work
```

Differential Revision: [D31447615](https://our.internmc.facebook.com/intern/diff/D31447615)

[ghstack-poisoned]

vkuzo committed:
Update on "Quantization docs: rewrite API reference to be more automated"


Description:

This PR switches the quantization API reference to use `autosummary`
for each section.  We define the sections and manually write a list
of modules/functions/methods to include, and sphinx does the rest.
A result is a single page where we have every quantization function
and module with a quick autogenerated blurb, and user can click
through to each of them for a full documentation page.

This mimics how the `torch.nn` and `torch.nn.functional` doc
pages are set up.

In detail, for each section before this PR:
* creates a new section using `autosummary`
* adds all modules/functions/methods which were previously in the manual section
* adds any additional modules/functions/methods which are public facing but not previously documented
* deletes the old manual summary and all links to it

Test plan:

```
cd docs
make html
python -m http.server
// renders well, links work
```

Differential Revision: [D31447615](https://our.internmc.facebook.com/intern/diff/D31447615)

[ghstack-poisoned]

vkuzo committed:
Update on "Quantization docs: rewrite API reference to be more automated"


Description:

This PR switches the quantization API reference to use `autosummary`
for each section.  We define the sections and manually write a list
of modules/functions/methods to include, and sphinx does the rest.
A result is a single page where we have every quantization function
and module with a quick autogenerated blurb, and user can click
through to each of them for a full documentation page.

This mimics how the `torch.nn` and `torch.nn.functional` doc
pages are set up.

In detail, for each section before this PR:
* creates a new section using `autosummary`
* adds all modules/functions/methods which were previously in the manual section
* adds any additional modules/functions/methods which are public facing but not previously documented
* deletes the old manual summary and all links to it

Test plan:

```
cd docs
make html
python -m http.server
// renders well, links work
```

Differential Revision: [D31447615](https://our.internmc.facebook.com/intern/diff/D31447615)

[ghstack-poisoned]

vkuzo commented:
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31447615).
